src/Util.cpp

```diff
@@ -42,9 +42,10 @@ void Util::AddToNodes(Nodes& nodes, const int_seg seg) {
 	nodes.xb.push_back(seg.xb);
 	nodes.yb.push_back(seg.yb);
 	nodes.zb.push_back(seg.zb);
-	nodes.phix.push_back(seg.phix);
-	nodes.phiy.push_back(seg.phiy);
-	nodes.phiz.push_back(seg.phiz);
+	// TODO::FEATURE
+	nodes.phix.push_back(1.0/seg.phix);
+	nodes.phiy.push_back(1.0/seg.phiy);
+	nodes.phiz.push_back(1.0/seg.phiz);
```

Comment:
Points of clarification, why the change to reciprocal of the variables and what is the TODO feature?

Reply:
Precomputing the reciprocal means that the reciprocal is only computed once (all points share the same nodes). Division is ~10x more costly than multiplication so it reduces the total compute cycles needed. "TODO" can be removed. Note that I didn't' test it out but it should give the same results.